daemons/controld/controld_schedulerd.c

```diff
@@ -45,19 +45,19 @@ controld_shutdown_schedulerd_ipc(void)
  * \internal
  * \brief Save CIB query result to file, raising FSA error
  *
- * \param[in] msg        Ignored
- * \param[in] call_id    Call ID of CIB query
- * \param[in] rc         Return code of CIB query
- * \param[in] output     Result of CIB query
- * \param[in] user_data  Unique identifier for filename (will be freed)
+ * \param[in]     msg        Ignored
+ * \param[in]     call_id    Call ID of CIB query
+ * \param[in]     rc         Return code of CIB query
+ * \param[in,out] output     Result of CIB query
```

Comment:
Where do we modify `output`? It can't be `const` but maybe `[in]`?

Reply:
It's a bit obscure, but getDocPtr() is in the call chain, and it will create a document node if one doesn't already exist

That prevents a bunch of the xml.c functions from using const. There are probably places where we can safely assume the document already exists and use xml->doc directly instead of calling getDocPtr(). It's likely even possible to always ensure that xml->doc exists as long as we created xml. But that would be a major pain to go through every caller.